### PR TITLE
[MetaSchedule] Relax conditions of rule Cross-Thread Reduction

### PR DESCRIPTION
**Pull request by MasterJH5574**
This PR relaxes the conditions of Meta-Schedule schedule rule CrossThreadReduction. The rules are previously a bit over-strict, and some workloads with small reduction loop length are unable to be optimized by cross-thread reduction automatically. In this PR, we relax the rules so that such workloads can be optimized.

cc @junrushao

### CODE REVIEW

File: tests/python/unittest/test_meta_schedule_schedule_rule_cross_thread_reduction.py [hunk 1]
```diff
@@ -663,8 +685,85 @@ def argmax_1(
...
+
+    mod = argmax_32
+    actual = ms.TuneContext(
+        mod=mod,
+        target=Target("nvidia/geforce-rtx-3090", host="llvm"),
+        space_generator=ms.space_generator.PostOrderApply(),
+        sch_rules=get_rules("cuda", ms.schedule_rule.CrossThreadReduction),
+        task_name="test",
+    ).generate_design_space()
+    print(actual[1].mod.script())
```
Comment (junrushao) [inline tests/python/unittest/test_meta_schedule_schedule_rule_cross_thread_reduction.py:755]: let's remove this debug printing

Comment (MasterJH5574) [reply tests/python/unittest/test_meta_schedule_schedule_rule_cross_thread_reduction.py:755]: Sorry my negligence 🤦‍♂️

Comment (MasterJH5574) [reply tests/python/unittest/test_meta_schedule_schedule_rule_cross_thread_reduction.py]: Done!

**Comment by junrushao** [comment]
The only question from me is: is it going to make the search space smaller if rfactor or cross-thread reduction is improperly applied? If not, we can get it in

**Comment by MasterJH5574** [comment]
> The only question from me is: is it going to make the search space smaller if rfactor or cross-thread reduction is improperly applied? If not, we can get it in

@junrushao This PR is going to make the search space larger for workloads which have small reduction loop length. Previously, the transformations (rfactor or cross-thread reduction) are not applied to these workloads at all. The transformations are not improperly applied.

**Review by junrushao — APPROVED** [review]
Thanks for the clarification! LGTM!

**Pull request merged**